File: apps/backend/app/schemas/paper.py
from pydantic import BaseModel

class PaperAuthor(BaseModel):
    name: str

class Paper(BaseModel):
    title: str
    summary: str
    pdf_url: str
    published: str
    authors: list[PaperAuthor]

class PaperSearchResponse(BaseModel):
    papers: list[Paper]

File: apps/frontend/types/paper.ts
export interface PaperAuthor {
  name: string;
}

export interface Paper {
  title: string;
  summary: string;
  pdf_url: string;
  published: string;
  authors: PaperAuthor[];
}

export interface PaperSearchResponse {
  papers: Paper[];
}

export interface IngestionResponse {
  total_chunks: number;

  sample_chunk: {
    chunk_id: string;
    text?: string;
    content?: string;
  };
}
